chain-impl-mockchain/src/fragment/content.rs

```diff
@@ -0,0 +1,82 @@
+use crate::fragment::Fragment;
+use crate::key::Hash;
+use chain_core::property::Serialize;
+use std::slice;
+
+pub type BlockContentHash = Hash;
+pub type BlockContentSize = u32;
+
+/// Block Contents
+///
+/// To create this structure, make a ContentsBuilder and use into()
+#[derive(Debug, Clone)]
+pub struct Contents(pub(super) Box<[Fragment]>);
+
+impl PartialEq for Contents {
+    fn eq(&self, rhs: &Self) -> bool {
+        self.compute_hash_size() == rhs.compute_hash_size()
+    }
+}
+impl Eq for Contents {}
+
+impl From<ContentsBuilder> for Contents {
+    fn from(content_builder: ContentsBuilder) -> Self {
+        Contents(content_builder.fragments.into())
+    }
+}
```

Comment:
consensus is to have `ContentsBuilder::build()` function instead

Reply:
will keep that in mind, definitely better